lib/data/src/text.rs

```diff
@@ -0,0 +1,52 @@
+use std::ops::Add;
+use std::ops::Sub;
+use serde::Serialize;
+use serde::Deserialize;
+
+#[derive(Clone,Copy,Debug,Default,PartialEq,Eq,PartialOrd,Ord,Serialize,Deserialize)]
+pub struct Index { pub value:usize }
+
+#[derive(Clone,Copy,Debug,Default,PartialEq,Eq,PartialOrd,Ord,Serialize,Deserialize)]
+pub struct Size { pub value:usize }
+
+#[derive(Clone,Copy,Debug,Default,PartialEq,Eq,PartialOrd,Ord,Serialize,Deserialize)]
+pub struct Span { pub index: Index, pub size: Size }
+
+impl Index {
+    pub fn new(value: usize) -> Self {
+        Index {value}
+    }
+}
+
+impl Size {
+    pub fn new(value: usize) -> Self {
+        Size {value}
+    }
+}
+
+impl Span {
+    pub fn new(index: Index, size: Size) -> Self {
+        Span {index,size}
+    }
+}
+
+impl Add for Size {
+    type Output = Size;
+    fn add(self, rhs:Size) -> Size {
+        Size {value:self.value+rhs.value}
```

Comment:
Missing doc comments. Please be sure you remove all compilation warnings and all linter warnings before comitting rust files.

Reply:
I don't see any compilation or linter warnings.

Reply:
`#![warn(missing_docs)]` wasn't enabled